### PR TITLE
Optionally allow range deletions for truncates.

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Optionally enable range deletions in the WAL for truncate operationns in the cluster, too. This can speed up truncate operations for large collections/shards. Range deletions are turned off by default to ensure compatibility with all versions of arangosync. They can be turned on by using the startup option `--rocksdb.use-range-delete-in-wal` and setting it to `true`. The default value of that option will change from `false` to `true` in ArangoDB 3.10.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [x] Backport for 3.9: this PR
  - [ ] Backport for 3.8: -
  - [ ] Backport for 3.7: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBCollection.cpp [hunk 1]
```diff
@@ -2103,11 +2098,14 @@ void RocksDBCollection::invalidateCacheEntry(RocksDBKey const& k) const {
 
 /// @brief can use non transactional range delete in write ahead log
 bool RocksDBCollection::canUseRangeDeleteInWal() const {
+  TRI_IF_FAILURE("ForceRangeDeleteInWal") { return true; }
   if (ServerState::instance()->isSingleServer()) {
-    // disableWalFilePruning is used by createIndex
-    return _numIndexCreations.load(std::memory_order_acquire) == 0;
+    return true;
```
Comment (cpjulia) [inline arangod/RocksDBEngine/RocksDBCollection.cpp:2103]: in single server, is it hardcoded to always use range deletion?

**Review by cpjulia — APPROVED** [review]
LGTM

File: arangod/RocksDBEngine/RocksDBCollection.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBCollection.cpp:2103]: Yes.

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,15 @@
+v3.9.3 (XXXX-XX-XX)
+-------------------
+
+* Optionally enable range deletions in the WAL for truncate operationns in
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Optionally enable range deletions in the WAL for truncate operations in 
```

**Pull request merged**